templates/batch-llm/batch_llm.py

```diff
@@ -0,0 +1,78 @@
+from vllm import LLM, SamplingParams
+from typing import Dict
+import numpy as np
+import ray
+import os
+
+# Set the Hugging Face token. Replace the following with your token.
+HF_TOKEN = os.environ.get("HF_TOKEN")
+if not HF_TOKEN:
+    raise ValueError("Please specify environment variable `HF_TOKEN` as Hugging Face token to access models in Hugging Face.")
+
+# Set to the model that you wish to use. Note that using the llama models will require a hugging face token to be set.
+HF_MODEL = "meta-llama/Llama-2-7b-chat-hf"
+
+# Read one text file from S3. Ray Data supports reading multiple files
+# from cloud storage (such as JSONL, Parquet, CSV, binary format).
+INPUT_PATH = "s3://anonymous@air-example-data/prompts.txt"
+
+# Initialize Ray with a Runtime Environment.
+ray.init(
+    runtime_env={
+        "env_vars": {"HF_TOKEN": HF_TOKEN},
+        "pip": ["vllm"],
+    }
+)
+
+# Create a sampling params object.
+sampling_params = SamplingParams(temperature=0)
+
+
+# Create a class to do batch inference.
+class LLMPredictor:
+    def __init__(self):
+        # Create an LLM.
+        self.llm = LLM(model=HF_MODEL)
```

Comment:
how does vllm know where to get model from?

Non blocking for V0: How about custom models. In platform serving lot of times users have finetuned or custom models in s3 they want to run inference on . How do we support that?

Reply:
> how does vllm know where to get model from?

vLLM always loads the model from Hugging Face - https://github.com/vllm-project/vllm/blob/main/docs/source/models/engine_args.rst . The model name has to be Hugging Face model name.

> Non blocking for V0: How about custom models. In platform serving lot of times users have finetuned or custom models in s3 they want to run inference on . How do we support that?

Yes good catch. Discussed offline, for V0, we can just support Hugging Face models. In V1/followup PR, let's tackle the custom/fine-tuned model, and change to use Anyscale vLLM fork.

Reply:
Discussed with @gvspraveen offline, using Endpoint docker image, which contains Anyscale LLM.